pkg/apis/core/types_shoot.go

```diff
@@ -706,6 +708,12 @@ type Worker struct {
 	Zones []string
 }
 
+// WorkerSystemComponents contains configuration for system components related to this worker pool
+type WorkerSystemComponents struct {
+	// Allow determines whether the pool should be allowed to host system components or not (defaults to true)
+	Allow *bool
```

Comment:
Why is this a pointer/an optional field? `WorkerSystemComponents` itself is already optional, so the `Allow` could be a regular field inside it. cc @gardener/gardener-maintainers wdyt?

Reply:
+1
No necessity to use a pointer.

Reply:
Removed